Improve website style on Arabic

ahmed0saber opened the pull request:
<!-- Add the number of the issue this pull request is closing here -->
Closes #251 

<!-- Description of the changes this pull request introduces -->
**What change does this pull request introduce?**
Improving the style of TheAlgorithms website on Arabic, by adjusting the direction to "rtl", replacing left and right of margin and padding to inline-start and inline-end, forcing "ltr" direction in some places, and finally adding some custom styling if required.

<!-- If applicable, include screenshots of the issue you solved -->
**Screenshots**
![screencapture-3000-thealgorithms-website-q2fhlgkhxd5-ws-eu110-gitpod-io-ar-2024-04-28-21_47_43](https://github.com/TheAlgorithms/website/assets/64714761/065c60a8-85ed-456f-bfd9-868a68c0fafe)

**Checklist**

- [x] I worked on a branch other than `main`.
- [x] My branch is up-to-date with the Upstream `main` branch.
- [x] I have fixed potential errors using `yarn lint`.
- [x] I ran `yarn build` to check everything still builds successfully.

ahmed0saber committed:
Improve home page style on arabic

ahmed0saber committed:
Improve search page style on arabic

ahmed0saber committed:
Improve algorithm page style on arabic

ahmed0saber committed:
Fix linting issues

maltejur commented on styles/globals.css:
```diff
@@ -129,3 +129,24 @@ html::-webkit-scrollbar {
+html[lang=ar] .AlgorithmCard__div nav,
+html[lang=ar] .AlgorithmCard__div h2 {
+  direction: ltr;
+}
```
Could you move the styles you added here to the respective `style.module.css` files instead?

ahmed0saber replied:
I have moved them.
Learnt something new today :)

maltejur reviewed (changes requested):
Thanks, this seems like a great change. One small thing.

ahmed0saber committed:
Move Arabic global styles to respective css modules

ahmed0saber committed:
Fix alignment of translateIcon in navbar side menu

ahmed0saber committed:
Fix linting issues

ahmed0saber commented:
Hello @maltejur , hope you are doing well.
I have moved the global styles of Arabic and RTL into respective CSS modules, and fixed a small issue I have found with the translate icon in side menu of navbar.

maltejur reviewed (approved):
Looks great now, thanks!
